implementations/micrometer-registry-signalfx/src/main/java/io/micrometer/signalfx/SignalFxMeterRegistry.java

```diff
@@ -104,7 +104,7 @@ else if ("https".equals(apiUri.getScheme())) {
 
     @Override
     protected void publish() {
-        final long timestamp = clock.wallTime();
+        final long timestamp = (clock.wallTime() / config.step().toMillis()) * config.step().toMillis();
```

Comment:
After thinking more about it, I'm wondering why we would change this. Before this change it is sending the timestamp as the time when publishing starts. Why would we change it to the start of the step with this change? That seems like a separate decision to make not directly related to the change of this pull request. I think I'll revert this change. Please open a separate issue or pull request to consider making this change.

Reply:
Earlier, the reporting was done at the top of the step (for simplicity let's assume 1min step). And if I understand right all the instances will report almost similar timestamps (i,e start of the minute). With this change, the timestamp will be de-synchronized, it will no more be the start of the minute, what I tried to do was to make it to the top of the minute.

Because when sending data from multiple instances without instanceId (or a unique identifier for an instance), the back-end of SignalFx might be able to calculate what is step a particular metric is reported at. Having said that, I will follow-up on this with a different PR after confirming with someone from SignalFx confirm this.